src/lib.rs

```diff
@@ -901,6 +912,9 @@ impl<TRepoTypes: RepoTypes> Future for IpfsFuture<TRepoTypes> {
                         let mut addresses = Vec::new();
                         addresses.extend(Swarm::listeners(&self.swarm).cloned());
                         addresses.extend(Swarm::external_addresses(&self.swarm).cloned());
+                        for addr in &mut addresses {
+                            addr.push(Protocol::P2p(self.swarm.ipfs().peer_id().into()));
+                        }
```

Comment:
Not saying this cannot be outside http, but I am not seeing why it needs to be in IpfsFuture *AND* have Ipfs::peer_id(); the peer_id could had been cloned inside the ipfsfuture but ... Not really seeing a point in that either.

Reply:
The `PeerId` could also be transmitted with the event, I guess that's the simplest alternative.